### PR TITLE
Set ttl to specific cache key for Redis Storage Adapter

### PR DESCRIPTION
**Pull request by pensiero**
Implemented two methods inside the `Storage\Adapter\Redis` that allow to:
- set a ttl to a specific key after setting it up (`setTimeout`)
- get the remaining tll of a specific key (`getRemainingTimeout`)

Actually is possible to specify a ttl to the adapter, but it will be assumed by all the keys that are setted up after the initialization.
Tests and interface implemented.

### CODE REVIEW

**Comment by Maks3w** [comment]
Please rebase against current master

**Comment by gianarb** [comment]
3f33aaf What fix? :dart:

**Comment by pensiero** [comment]
:angry: it was something from another pull request.. damn xD i'll rebase

File: src/Storage/ExpirableInterface.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Cache\Storage;
+
+interface ExpirableInterface
```
Comment (Maks3w) [inline src/Storage/ExpirableInterface.php]: Please add a docblock describing the purpose of the interface. Like an introduction paragraph

**Comment by pensiero** [comment]
How can I do it? The master branch of my fork contain that commit (that is used for this PR: #3)

File: src/Storage/ExpirableInterface.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ *  and get the remaining ttl applied to their items
+ *
+ * @author Konstantin Kudryashov <ever.zet@gmail.com>
+ */
+interface ExpirableInterface
+{
+    /**
+     * Sets an expiration date (a timeout) on an item
+     *
+     * @param $key
```
Comment (Maks3w) [inline src/Storage/ExpirableInterface.php]: Improve params and return types

**Comment by Maks3w** [comment]
Rebase or undo the cosmetic changes. Try to keep your commits focused in the issue and don't change unrelated lines.

**Comment by pensiero** [comment]
@Maks3w how can I do it?

**Comment by pensiero** [comment]
PS: sorry for that, it was not intentional :(

**Comment by pensiero** [comment]
Is it ok now?

**Comment by marc-mabe** [comment]
1.) What is the difference of `touchItem` and `setTimeout`. It's currently not implemented in the redis adapter and falls back to https://github.com/zendframework/zend-cache/blob/master/src/Storage/Adapter/AbstractAdapter.php#L1081
2.) `getRemainingTimeout` should be part of `getMetadata`

**Comment by pensiero** [comment]
1) `touchItem` reset the lifetime of an Item, while `setTimeout` allows to specify a different lifetime than the global one (specificied using `setTtl`)
2) How can I achieve this? I saw that ìnternalGetMetadata` return an empty array (https://github.com/zendframework/zend-cache/blob/master/src/Storage/Adapter/AbstractAdapter.php#L591-L598) ... is it correct?

**Comment by marc-mabe** [comment]
> 1) touchItem reset the lifetime of an Item, while setTimeout allows to specify a different lifetime than the global one (specificied using setTtl)

The way to go is the following if you have to change the TTL to something different as the current:
```php
$cache->getOptions()->setTtl($ttl);
$cache->touchItem($key);
```
So you should overwrite `internalTouchItem` and set the TTL from options

> 2) How can I achieve this? I saw that ìnternalGetMetadata` return an empty array (https://github.com/zendframework/zend-cache/blob/master/src/Storage/Adapter/AbstractAdapter.php#L591-L598) ... is it correct?

Like it's done on MongoDB:
https://github.com/zendframework/zend-cache/blob/master/src/Storage/Adapter/MongoDb.php#L261-L266
https://github.com/zendframework/zend-cache/blob/master/src/Storage/Adapter/MongoDb.php#L241-L243

**Comment by pensiero** [comment]
Moved to #10

**Pull request closed without merging**